### PR TITLE
doc: add checklist for new device plugins

### PR DESCRIPTION
**Pull request by mythi**
Closes: #520

### CODE REVIEW

File: DEVEL.md [hunk 1]
```diff
@@ -122,6 +122,20 @@ Otherwise, they can be logged as simple values:
...
 
+### Checklist for new device plugins
+
+For new device plugins contributed to this repository, below is a
+checklist to get the plugin on par feature and quality wise with
+others:
+
+1. Plugin binary available in `cmd/`, its corresponding Dockerfile in `build/docker/` and deployment Kustomization/YAMLs in `deployments/`.
+2. Plugin binary Go unit tests implemented and passing with >80% coverage: `make test WHAT=./cmd/<device>`.
+3. Plugin binary linter chcks passing: `make lint`.
```
Comment (msivosuo) [inline DEVEL.md:133]: chcks -> checks

File: DEVEL.md [hunk 2]
```diff
@@ -122,6 +122,20 @@ Otherwise, they can be logged as simple values:
...
+4. Plugin e2e tests implemented and passing: `go test -v ./test/e2e/... -args -ginkgo.focus "<DEVICE>"`.
+5. Plugin CRD API added to `pkg/apis/deviceplugin/v1` and CRDs generated: `make generate`.
+6. Plugin CRD validation tests implemented in `test/envtest/` and passing: `make envtest`.
+7. Plugin CRD controller implemented in `pkg/controllers/` and added to the manager in `cmd/operator/main.go`.
```
Comment (msivosuo) [inline DEVEL.md:137]: Should we have an item about corresponding documentation too?

File: DEVEL.md (hunk 1 above)
Comment (mythi) [reply DEVEL.md]: fixed

File: DEVEL.md (hunk 2 above)
Comment (mythi) [reply DEVEL.md]: good comment, added

File: DEVEL.md [hunk 3]
```diff
@@ -122,6 +122,21 @@ Otherwise, they can be logged as simple values:
...
 ```
 
+### Checklist for new device plugins
+
+For new device plugins contributed to this repository, below is a
+checklist to get the plugin on par feature and quality wise with
+others:
+
+1. Plugin binary available in [`cmd/`](cmd), its corresponding Dockerfile in [`build/docker/`](build/docker) and deployment Kustomization/YAMLs in [`deployments/`](deployments).
+2. Plugin binary Go unit tests implemented and passing with >80% coverage: `make test WHAT=./cmd/<device>`.
```
Comment (bart0sh) [inline DEVEL.md:132]: Would it be better to use <plugin> instead of <device> here?

Comment (mythi) [reply DEVEL.md]: changed to `<plugin>`

File: DEVEL.md [hunk 4]
```diff
@@ -122,6 +122,21 @@ Otherwise, they can be logged as simple values:
...
+others:
+
+1. Plugin binary available in [`cmd/`](cmd), its corresponding Dockerfile in [`build/docker/`](build/docker) and deployment Kustomization/YAMLs in [`deployments/`](deployments).
+2. Plugin binary Go unit tests implemented and passing with >80% coverage: `make test WHAT=./cmd/<plugin>`.
+3. Plugin binary linter checks passing: `make lint`.
+4. Plugin e2e tests implemented in [`test/e2e/`](test/e2e) and passing: `go test -v ./test/e2e/... -args -ginkgo.focus "<plugin>"`.
+5. Plugin CRD API added to [`pkg/apis/deviceplugin/v1`](pkg/apis/deviceplugin/v1) and CRDs generated: `make generate`.
+6. Plugin CRD validation tests implemented in [`test/envtest/`](test/envtest) and passing: `make envtest`.
+7. Plugin CRD controller implemented in [`pkg/controllers/`](pkg/controllers) and added to the manager in `cmd/operator/main.go`.
+8. Plugin documentation written `cmd/<device>/README.md` and optionally end to end demos created in [`demo`](demo).
```
Comment (rojkov) [inline DEVEL.md:138]: better replace `<device>` with `<plugin>` here too for consistency.

Comment (mythi) [reply DEVEL.md]: oops, fixed

**Review by bart0sh — APPROVED** [review]
/lgtm

**Review by rojkov — APPROVED** [review]

**Pull request merged**